src/EditorFeatures/CSharpTest/CodeActions/SyncNamespace/CSharpSyncNamespaceTestsBase.cs

```diff
@@ -0,0 +1,205 @@
+                    {
+                        // the original source document does not exist in the new solution.
+                        var oldSolution = result.Item1;
+                        var newSolution = result.Item2;
+
+                        Assert.Null(newSolution.GetDocument(oldDocumentId));
+
+                        var newDocument = GetDocumentToVerify(expectedChangedDocumentId: null, oldSolution, newSolution);
+                        actualFolderPaths.Add(string.Join(PathUtilities.DirectorySeparatorStr, newDocument.Folders));
+                    }
+
+                    Assert.True(expectedFolderPaths.Count() == actualFolderPaths.Count, "Number of available \"Move file\" actions are not equal.");
+                    foreach (var expected in expectedFolderPaths)
+                    {
+                        Assert.True(actualFolderPaths.Contains(expected));
+                    }
+                }
+                else
+                {
+                    var (actions, _) = await GetCodeActionsAsync(workspace, testOptions);
+                    if (actions.Length > 0)
+                    {
+                        var renameFileAction = actions.Any(action => action is CSharpSyncNamespaceService.MoveFileCodeAction);
+                        Assert.False(renameFileAction, "Rename File to match type code action was not expected, but shows up.");
+                    }
+                }
+            }
+
+            async Task<List<Tuple<Solution, Solution>>> TestOperationAsync(
+                TestParameters parameters,
+                TestWorkspace workspace,
+                string expectedCode)
+            {
+                var results = new List<Tuple<Solution, Solution>>();
+
+                var (actions, _) = await GetCodeActionsAsync(workspace, parameters);
+                var moveFileActions = actions.Where(a => a is CSharpSyncNamespaceService.MoveFileCodeAction);
+
+                foreach (var action in moveFileActions)
+                {
+                    var operations = await action.GetOperationsAsync(CancellationToken.None);
+
+                    results.Add(
+                        await TestOperationsAsync(workspace,
+                        expectedText: expectedCode,
+                        operations: operations,
+                        conflictSpans: ImmutableArray<TextSpan>.Empty,
+                        renameSpans: ImmutableArray<TextSpan>.Empty,
+                        warningSpans: ImmutableArray<TextSpan>.Empty,
+                        navigationSpans: ImmutableArray<TextSpan>.Empty,
+                        expectedChangedDocumentId: null));
+                }
+                return results;
+            }
+        }
+
+        protected async Task TestChangeNamespaceAsync(
+            string initialMarkUp,
+            string expectedSourceOriginal,
+            string expectedSourceReference = null,
+            bool hasTwoProjects = false)
```

Comment:
it seems weird to have to say hasTwoProjects in the parameter list.  isn't that implied by the markup you get passed in?

Reply:
fixed